File: app/src/main/java/mitya/yahnc/ui/TopStoriesFragment.java
package mitya.yahnc.ui;

import mitya.yahnc.network.StoryService;
import rx.Observable;
import rx.schedulers.Schedulers;

/**
 * Created by Mitya on 25.08.2016.
 */
public class TopStoriesFragment extends RemoteStoryFragment {
    public TopStoriesFragment() {
        super(StoryService.getInstance().getTopStories().
                flatMap(stories -> Observable.from(stories).subscribeOn(Schedulers.io())));
    }
}

File: app/src/main/java/mitya/yahnc/ui/TopStoriesFragment.kt
package mitya.yahnc.ui

import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import mitya.yahnc.network.HnService

/**
 * Created by Mitya on 25.08.2016.
 */
class TopStoriesFragment : RemoteStoryFragment() {
    override fun getCurrentStories(): Observable<Int> {
        return HnService.getTopStories().flatMap { stories -> Observable.fromArray(*stories).subscribeOn(Schedulers.io()) }
    }
}
